spark/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -1184,6 +1184,30 @@ class DeltaSuite extends QueryTest
     }
   }
 
+  test("streaming write to catalog table via data source API") {
+    withTable("tab") {
+      Seq(1, 2, 3).toDF.write.format("delta").saveAsTable("tab")
+      checkDatasetUnorderly(spark.read.table("tab").as[Int], 1, 2, 3)
+
+      // Streaming write should correctly update the Delta table.
+      val input = MemoryStream[Int]
+      val q = input.toDF
+        .writeStream
+        .format("delta")
+        .option(
+          "checkpointLocation",
+          Utils.createTempDir(namePrefix = "tahoe-test").getCanonicalPath)
+        .toTable("tab")
```

Comment:
what fails in this test without the changes in the rule?

Reply:
Nothing, now that you mention -- I couldn't figure out how to hook in a transaction post-commit hook, and forgot that I ended up just using the test as a debugger entry point to verify that the new rule triggered.

I figured out a way to access the plan tho.